### ISSUE TITLE
Process - Expedited merging of event page changes

### ISSUE THREAD

**Issue by sweco-sedalh** [qgis/QGIS-Website#491]
### Feature Description

Not quite sure if this is an appropriate venue for this, or if it should be a QEP or similar, but I'll start here.

---

For event pages (as the UC 2025 pages just merged) there are usually three things true - there'll usually be one or two people responsible for updating them from the LOC (local organizing committee), other potential reviewers likely won't be able to tell the validity of the information (as the PR might be the first time they see it) and there commonly is an urgency (especially the days before/during the event, where changes need to get out ASAP) that's not applicable for most other changes.

Therefore, I think it would make sense for an expedited PR process. Something along these lines:

> LOCs can appoint 1-2 persons who get permission to merge PRs, including their own, touching only the specific event pages. PRs touching other parts require the usual review process.

I don't think that Github supports branch policies scoped to directories (not even with a CODEOWNERS file), so this would have to be a "soft" policy (i.e. the appointed persons get the ability to merge any PRs, but must promise to follow the process).

**Comment by timlinux** [comment]
Hi @sweco-sedalh thank you for your ticket. For me I prefer that we work through @Xpirix to manage merging PRs so that he can maintain review of everything coming in to the web site. If you follow the same approach as UC2024, you can establish your own event specific site (or just use the QGIS wiki - https://github.com/qgis/QGIS/wiki/) and then independently manage the content there.  @Xpirix is very responsive and should be able to merge any PR's within a short period after it is opened.

**Comment by sweco-sedalh** [comment]
The current idea - having the UC-site as part of qgis.org - was the result of a discussion at the CM in Bratislava. Mainly it's about reducing the amount of work the LOC has to do (now, the following conferences just have to adjust the date, location and other details and send a PR, as opposed to setting up hosting (even if Github Pages makes it reasonably easy), DNS, syncing styling with QGIS at large, etc.). Secondarily, this can also help with visibility for the conference.

While I in no way doubt that @Xpirix does and will do his best, we don't want to put the additional burden/pressure of quickly (as in, within hours even outside his normal workhours, if there's something important shortly before or during the conference) reviewing changes.

I don't think splitting the information between more places is a good idea, so if this is possible we might want to reconsider how the conference pages are handled.

**Comment by timlinux** [comment]
hi @sweco-sedalh 

Thank you by the way for organising the UC2025 and please don't take me to be obstructionist,  but I am wary about putting all of the UC event content into the QGIS web site. We spent a lot of time  trying to prune the site down to a more carefully curated set of content when we did the web site overhaul and if feels like hosting the UC2025 pages there will take us in the opposite direction.

Some solutions I would prefer:

1. Use the wiki - low effort and we used it for many years to organise events.
2. We could give you a separate subdomain e.g. uc2025.qgis.org and ask @Xpirix to create a blank copy of our hugo template there. Then you can manage content there as you see fit and we can take it offline again after the UC
3. Consider finding a tool for managing the conference e.g. https://getindico.io (just a randome google link, not sure how good it is) that could be used year on year so we can solve this over the longer term
4. If others don't agree with me about my position  above we could perhaps use git submodules to delegate responsibility of some pages to you...

**Comment by mbernasocchi** [comment]
@timlinux i totally agree. The page has went through a big diet and we should keep it this way.

A conference page under qgis.org is a gret idea, but it should be just the entry point to another page with all the detailed information the LOC manages. I think, ideally we should have a template for a subdomain for the conference so that we use something a bit better looking than the wiki and has the same look and feel to qgis.org.
This would take the user conference much closer to qgis.org and still allow the organisers to be very quick in adapting things.

The main entry page should have just dates, location, a link to a program and a big button to register and one to go to the uc2025.qgis.org page 😀

**Comment by timlinux** [comment]
@sweco-sedalh so if it is good for you I will ask @Xpirix to make a template for UC events and set up a gh repo for the upcoming 2025 event with the template and tooling needed to publish it to uc2025.qgis.org.

We will use the same template as qgis.org for header and footer and components and you can add content there as you wish. Maybe we can have some standard pages so that each conference (2026,27 etc) has the same layout just with event specific data...

**Comment by Xpirix** [comment]
@sweco-sedalh  I am eager to assist in establishing a template repository for the QGIS UC 2025.
Please let us know your thoughts on this.

**Comment by karl-magnus** [comment]
Hi all! Probably a good idea if you say so. The only thing I am worried about is that the event page is already created, published and announced. It is not fun to redo things or mess up the communication with inconsistent references to different pages. But this is maybe not a problem and can be solved.

**Comment by timlinux** [comment]
Hi @karl-magnus 

Thank you for your message. I just want to clarify that we are not proposing to remove the single event page that is already published. Rather we are proposing  that we provide you with a subdomain that you can use to publish any additional information. We will make a github repo for you and set up a theme and automations similar to what we have for the QGIS.org site and from there you can completely manage the content. We will give you any support you need.

From the existing event page (https://qgis.org/community/organisation/meetings/uc-2025/) we add a link to the event subdomain (https://uc2025.qgis.org) and then you have carte blanche to manage the content there as needed.

I hope that clarifies things!

Regards

Tim

**Comment by karl-magnus** [comment]
Hi @timlinux 
OK then. Thanks! I think @sweco-sedalh will take care of the technical/practical when he is back from vacation. 
Regards

**Comment by Xpirix** [comment]
Hi @karl-magnus @sweco-sedalh and @timlinux 

I have created a new repository (on my GH account for now) based on the QGIS Website one. For now, I've filled it with the basic information from the UC 2025 page (on QGIS.org). Please feel free to check it and share your feedback.

- Repository: https://github.com/Xpirix/QGIS-UC-Website
- Github pages preview: https://xpirix.github.io/QGIS-UC-Website/

**Comment by sweco-sedalh** [comment]
Hi! Sorry for going MIA, was a lot to do in december at work.

I'm still worried about the extra burden this would place on you @Xpirix, as the two repos need to be kept in sync. I've not worked a lot with Hugo, is there some way to abstract the common parts (such as the theme) out (maybe add QGIS-Website as a git submodule to QGIS-UC-Website)?

But moving on with this solution, that would mean that the following needs to be done:

- [x] Move the new repo to the QGIS organization
- [x] Setup CI/CD, hosting and domain name
- [ ] Replace the current pages on QGIS-Website by a redirect to the new domain
- [x] Give me (and possibly @karl-magnus though I'm not sure how much you'll work directly on the website, you have enough else on your plate) write access to the QGIS-UC-Website repo

I can look into the third point once the new site is up. Can @Xpirix and @timlinux take care of the rest?

**Comment by Xpirix** [comment]
Hi @sweco-sedalh, thanks for the feedback. I will discuss these parts with @timlinux and get back to you.

**Comment by Xpirix** [comment]
> I'm still worried about the extra burden this would place on you @Xpirix, as the two repos need to be kept in sync. I've not worked a lot with Hugo, is there some way to abstract the common parts (such as the theme) out (maybe add QGIS-Website as a git submodule to QGIS-UC-Website)?

No worries, it is always a pleasure for me to assist you with it. Yes, the theme is a common part of these two repositories. However, we hardly update the theme so I don't think we need a periodic synchronization on that part. Also, we used a submodule for the `hugo-bulma-block-themes` before but decided to make it part of the repository as it was pretty complex to manage. We can always try to re-implement this later if needed.

After discussing with @timlinux:

>  Move the new repo to the QGIS organization

He will create a new repo (or eventually transfer my repo) on the QGIS organization and give you write access to it.

> Setup CI/CD, hosting and domain name

Sure, we will set up the server and the domain name. I will also set up the CI/CD once the server is ready.

> Give me (and possibly @karl-magnus though I'm not sure how much you'll work directly on the website, you have enough else on your plate) write access to the QGIS-UC-Website repo

Sure. Also, I'm happy to continue working on the template and set up the structure (second navigation menu and blank content for each page) following the UC 2024 website. What do you think?

**Comment by Xpirix** [comment]
Hello @mbernasocchi 

I have created a new repository on my GitHub account (https://github.com/Xpirix/QGIS-UC-Website) for the QGIS UC 2025 website. Would it be possible to transfer this repository to the QGIS organization and grant write access to @sweco-sedalh and @karl-magnus?

If this is feasible, please let me know when you're available, and I will transfer the repository to you.

**Comment by mbernasocchi** [comment]
Transfer is done. Permissions are added as requested.

Thanks   @Xpirix!

**Comment by Xpirix** [comment]
Hi everyone, Cc @sweco-sedalh @karl-magnus 

The website https://uc2025.qgis.org/ is now live with the `main` branch of https://github.com/qgis/QGIS-UC-Website.
For now, the deployment is done through a scheduled service that is triggered hourly.
